src/main/java/com/mykytaaa/cognito/service/SignUpUserService.java

```diff
@@ -0,0 +1,89 @@
+package com.mykytaaa.cognito.service;
+
+import com.mykytaaa.cognito.rest.dto.UserCognitoRequestDto;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import software.amazon.awssdk.services.cognitoidentityprovider.CognitoIdentityProviderClient;
+import software.amazon.awssdk.services.cognitoidentityprovider.model.AttributeType;
+import software.amazon.awssdk.services.cognitoidentityprovider.model.CognitoIdentityProviderException;
+import software.amazon.awssdk.services.cognitoidentityprovider.model.SignUpRequest;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class SignUpUserService {
+
+    private final CognitoIdentityProviderClient client;
+
+    @Value("${aws.client-id}")
+    private String clientId;
+
+    @Value("${aws.secret-key}")
+    private String secretKey;
+
+    public Boolean signUp(UserCognitoRequestDto requestDto) {
+        AttributeType attributeType = AttributeType.builder()
+                .name("email")
+                .value(requestDto.email())
+                .build();
+
+        List<AttributeType> attrs = new ArrayList<>();
+        attrs.add(attributeType);
+        try {
+            String secretVal = calculateSecretHash(clientId, secretKey, requestDto.email());
+            SignUpRequest request = SignUpRequest.builder()
+                    .userAttributes(attrs)
+                    .username(requestDto.email())
+                    .password(requestDto.password())
+                    .clientId(clientId)
+                    .secretHash(secretVal)
+                    .build();
+
+            client.signUp(request);
+            return true;
+        } catch (CognitoIdentityProviderException e) {
+            System.err.println(e.awsErrorDetails().errorMessage());
+        } catch (NoSuchAlgorithmException | InvalidKeyException e) {
+            throw new RuntimeException(e);
+        }
+        throw new RuntimeException();
+    }
+
+    public static String calculateSecretHash(String userPoolClientId, String userPoolClientSecret, String email) throws NoSuchAlgorithmException, InvalidKeyException {
```

Comment:
Why does static public method exist here while it may be re-used for many places while working with AWS Cognito?